### PR TITLE
fix: Correct error message when pypi-token.pypi has incorrect number of arguments.

### PR DESCRIPTION
**Pull request by radoshi**
# Pull Request Check List

Resolves: #8503

- [x] Added **tests** for changed code.
- [ ] Updated **documentation** for changed code.

### CODE REVIEW

File: src/poetry/console/commands/config.py [hunk 1]
```diff
@@ -138,6 +138,8 @@ def handle(self) -> int:
 
                 self.line(str(value))
             else:
+                if setting_key.split(".")[0] in self.LIST_PROHIBITED_SETTINGS:
+                    raise ValueError(f"Expected a value for {setting_key} setting.")
```
Comment (radoering) [inline src/poetry/console/commands/config.py:142]: Can we move this up, so we don't have to do the regex matching, which should not be relevant in this case?

File: tests/console/commands/test_config.py [hunk 2]
```diff
@@ -402,6 +402,16 @@ def test_set_pypi_token_unsuccessful_multiple_values(
     assert str(e.value) == "Expected only one argument (token), got 2"
 
 
+def test_set_pypi_token_no_values(
+    tester: CommandTester,
+) -> None:
+    print("Running this test")
```
Comment (radoering) [inline tests/console/commands/test_config.py:408]:
```suggestion
```

Comment (radoshi) [reply tests/console/commands/test_config.py]: Doh. Silly. Fixed.

File: src/poetry/console/commands/config.py (hunk 1 above)
Comment (radoshi) [reply src/poetry/console/commands/config.py]: Yeah good point. It'll make the code cleaner. Fixed.

**Review by radoering — APPROVED** [review]

**Pull request merged**